packages/@jsii/kernel/src/kernel.ts

```diff
@@ -1325,6 +1304,44 @@ export class Kernel {
     return property;
   }
 
+  /**
+   * Shared (non-public implementation) to as not to break API recording.
+   */
+  private _getBinScriptCommand(
+    req: api.GetScriptCommandRequest,
+  ): api.GetScriptCommandResponse {
+    const packageDir = this._getPackageDir(req.assembly);
+    if (fs.pathExistsSync(packageDir)) {
+      // module exists, verify version
+      const epkg = fs.readJsonSync(path.join(packageDir, 'package.json'));
+
+      if (!epkg.bin) {
+        throw new JsiiFault(
+          'There is no bin scripts defined for this package.',
+        );
+      }
+
+      const scriptPath = epkg.bin[req.script];
+
+      if (!epkg.bin) {
+        throw new JsiiFault(`Script with name ${req.script} was not defined.`);
+      }
```

Comment:
I would consider those two failure modes to be the same one, actually...

```suggestion
      const scriptPath = epkg.bin?.[req.script];

      if (!epkg.bin) {
        throw new JsiiFault(`Script with name ${req.script} was not defined.`);
      }
```